### PR TITLE
Implement CWD functionality

### PR DESCRIPTION
**Pull request by MaxVerevkin**
Enables `std::env::set_current_dir` and `std::env::current_dir`.

Also add `std::os::psp` module mainly for `OsStrExt`/`OsStringExt`, because there was no way to create `OsStr`/`OsString`.

I added `#[unstable(feature = "psp_std", issue = "none")]` to everything in that module because something must have been added.

### CODE REVIEW

File: library/std/src/sys/psp/os.rs [hunk 1]
```diff
@@ -1,10 +1,13 @@
 use super::{unsupported, Void};
 use crate::error::Error as StdError;
-use crate::ffi::{OsStr, OsString};
+use crate::ffi::{CString, OsStr, OsString};
 use crate::fmt;
 use crate::io;
+use crate::os::psp::ffi::OsStrExt;
 use crate::path::{self, PathBuf};
 
+static mut CWD: Option<CString> = None;
```
Comment (MaxVerevkin) [inline library/std/src/sys/psp/os.rs:9]: AFAIK there is no way to ask for cwd, so it's just stored here. PSPSDK does something similar.

File: library/std/src/os/mod.rs [hunk 2]
```diff
@@ -135,6 +135,9 @@ pub mod macos;
 pub mod netbsd;
 #[cfg(target_os = "openbsd")]
 pub mod openbsd;
+#[cfg(target_os = "psp")]
+#[unstable(feature = "psp_std", issue = "none")]
```
Comment (overdrivenpotato) [inline library/std/src/os/mod.rs:139]: Is `#[unstable]` required? The PSP is a tier 3 target so I think we don't need this at all.

Comment (MaxVerevkin) [reply library/std/src/os/mod.rs:139]: _Some_ stability attribute is required for all public `std` items. Marking it as stable feels wrong.

Comment (MaxVerevkin) [reply library/std/src/os/mod.rs:139]: But I can move it into `psp/mod.rs`

Comment (overdrivenpotato) [reply library/std/src/os/mod.rs:139]: Oh I see, well if it's required then this LGTM. Thanks!

**Pull request merged**